### PR TITLE
ExpandURL: Switch to Expandurl.com API

### LINKED ISSUE #2605: ExpandURL: API Gone offline - Update to new API: ExpandURL.com
It looks like http://longurl.org is offline -- For now I've turned off this IA. We need to determine if the API is offline permanently and if so, find an alternative or remove the Instant Answer until we have an API.   

------
IA Page:  http://duck.co/ia/view/expand_url 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @adman

### PR DESCRIPTION
**Pull request by sahildua2305**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [ ] New Instant Answer: **`New {IA Name} Instant Answer`**
- [x] Improvement
    - [x] Bug fix: **`ExpandURL: Fix #2605`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{SpiceRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes
Switched to expandurl.com API since longurl.com API is down.

###### Which issues (if any) does this fix?

Fixes #2605  

###### People to notify (@mention interested parties)
@moollaza 

---

Instant Answer Page: https://duck.co/ia/view/expand_url

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @Adman

### CODE REVIEW

**Comment by daxtheduck** [comment]
@sahildua2305 Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Spice/ExpandURL.pm [hunk 1]
```diff
@@ -6,7 +6,7 @@ use DDG::Spice;
 
 triggers start => "http", "https", "expand", "expandurl", "unshorten";
 
-spice to => 'http://api.longurl.org/v2/expand?url=$1&format=json&callback={{callback}}';
+spice to => 'http://expandurl.com/api/v1/?url=$1&format=json';
```
Comment (gaulrobe) [inline lib/DDG/Spice/ExpandURL.pm]: Since this new API doesn't have a `callback` do we need to add `spice wrap_jsonp_callback => 1;` here?

Comment (sahildua2305) [inline lib/DDG/Spice/ExpandURL.pm]: I am not sure about this one. Created the PR for getting feedback.

Comment (gaulrobe) [inline lib/DDG/Spice/ExpandURL.pm]: Try it out :smile: it sounds promising per the doc.

http://docs.duckduckhack.com/backend-reference/spice-attributes.html

**Comment by sahildua2305** [comment]
@gaulrobe Check now. I have set it to true. It still shows `406 Not Acceptable`. Can you please test it?

**Comment by gaulrobe** [comment]
That endpoint returns JSON, but sets the Content-Type as `text/html` in the response header... yuck! @moollaza is there a way to get around this?

**Comment by sahildua2305** [comment]
Yes, exactly what I mentioned here - https://github.com/duckduckgo/zeroclickinfo-spice/issues/2605#issuecomment-225052588

**Comment by sahildua2305** [comment]
@gaulrobe @moollaza Check this API - http://expand-url.herokuapp.com/expand?url=http://bit.ly/1U9JhGk
I am maintainer of this API. Let me know if I can switch this IA on this API.

**Comment by edgesince84** [comment]
@sahildua2305 thanks, we'll check it out

**Comment by moollaza** [comment]
> I am maintainer of this API. Let me know if I can switch this IA on this API.

@sahildua2305 we ask that developers don't make their own APIs for instant answers, unless they plan to open it to the public, and in the case of Heroku and similar, move it off a free tier which requires forced API downtime, meaning the IA may not work at times or provide very slow responses.

We do this because we don't want developers on the hook financially or otherwise for APIs that only DuckDuckGo is using. As well, thought we know our devs have the best intentions in mind, we try to use API's that we can reasonably expect to be around for the foreseeable future, being about 2 years.

There's actually an open PR for this IA that uses another API, untiny.me which looks like it should work: #2610

**Comment by sahildua2305** [comment]
@moollaza I totally understand that. I was actually thinking of shifting it off the free plan to the hobby plan (which has no forced freetime). And having this IA based on my API will act as another motivation for doing so.
Anyway if it's already working using untiny.me, should I close this PR? Why is that PR not merged. It's been put on hold. Any issues with that?

**Comment by moollaza** [comment]
> should I close this PR? 

Yes please, we can re-open if needed. Don't delete your branch!

> Why is that PR not merged. It's been put on hold. Any issues with that?

It looks like it was deployed but we were having issues with the API response on Beta. I've redeployed to see if the problem persists.

**Pull request closed without merging**